### PR TITLE
Make toolStrip non-nullable in ToolStripGripRenderEventArgs and ToolStripRenderEventArgs

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Make toolStrip non-nullable in ToolStripGripRenderEventArgs and ToolStripRenderEventArgs.
- Addresses feedback in https://github.com/dotnet/winforms/pull/6391.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6484)

### CODE REVIEW

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderEventArgs.cs [hunk 1]
```diff
@@ -13,8 +13,8 @@ public class ToolStripRenderEventArgs : EventArgs
         /// <summary>
         ///  This class represents all the information to render the toolStrip
         /// </summary>
-        public ToolStripRenderEventArgs(Graphics g, ToolStrip? toolStrip)
-            : this(g, toolStrip, new Rectangle(Point.Empty, toolStrip?.Size ?? Size.Empty), Color.Empty)
+        public ToolStripRenderEventArgs(Graphics g, ToolStrip toolStrip)
+            : this(g, toolStrip, new Rectangle(Point.Empty, toolStrip?.Size ?? throw new ArgumentNullException(nameof(toolStrip))), Color.Empty)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderEventArgs.cs:17]: Would this work?
```suggestion
            : this(g, toolStrip, new Rectangle(Point.Empty, toolStrip.OrThrowIfNull().Size), Color.Empty)
```

**Review by RussKie — COMMENTED** [review]
👍

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderEventArgs.cs (hunk 1 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderEventArgs.cs]: Ah, yes, I forgot I could also apply this here.

**Comment by dreddy-work** [comment]
@Olina-Zhang , can you guys run any Toolstrip grip scenarios here with these changes?

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang , can you guys run any Toolstrip grip scenarios here with these changes?

Will test this change tomorrow and update it in here.

**Comment by Olina-Zhang** [comment]
No new issue found when defined a custom ToolStrip with calling ToolStripRenderEventArgs and ToolStripGripRenderEventArgs, refer to https://docs.microsoft.com/en-us/dotnet/desktop/winforms/controls/how-to-implement-a-custom-toolstriprenderer?view=netframeworkdesktop-4.8, and also tested ToolStrip with different RenderModes.

**Pull request merged**